modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/schema/SchemaBuilder.java

```diff
@@ -0,0 +1,812 @@
+    {
+        name = hyphenize(name);
+
+        BigInteger minOccurs = required ? BigInteger.ONE : BigInteger.ZERO;
+        String collectionName = hyphenize(NameUtils.singularize(name));
+        LocalComplexType collectionComplexType = generateCollectionComplexType(collectionName, description, type);
+
+        TopLevelElement collectionElement = new TopLevelElement();
+        collectionElement.setName(name);
+        collectionElement.setMinOccurs(minOccurs);
+        collectionElement.setMaxOccurs("1");
+        collectionElement.setAnnotation(createDocAnnotation(description));
+        all.getParticle().add(objectFactory.createElement(collectionElement));
+
+        collectionElement.setComplexType(collectionComplexType);
+    }
+
+    private LocalComplexType generateCollectionComplexType(String name, final String description, final DataType type)
+    {
+        final LocalComplexType collectionComplexType = new LocalComplexType();
+        final ExplicitGroup sequence = new ExplicitGroup();
+        collectionComplexType.setSequence(sequence);
+
+        final TopLevelElement collectionItemElement = new TopLevelElement();
+        collectionItemElement.setName(name);
+        collectionItemElement.setMinOccurs(BigInteger.ZERO);
+        collectionItemElement.setMaxOccurs(SchemaConstants.UNBOUNDED);
+
+        final DataType genericType = getGenericType(type);
+        genericType.getQualifier().accept(new BaseDataQualifierVisitor()
+        {
+
+            @Override
+            public void onPojo()
+            {
+                collectionItemElement.setComplexType(newLocalComplexTypeWithBase(genericType, description));
+            }
+
+            @Override
+            protected void defaultOperation()
+            {
+                collectionItemElement.setComplexType(generateComplexValueType(genericType));
+            }
+        });
+
+        sequence.getParticle().add(objectFactory.createElement(collectionItemElement));
+
+        return collectionComplexType;
+    }
+
+    private LocalComplexType generateComplexValueType(DataType type)
+    {
+        LocalComplexType complexType = new LocalComplexType();
+        SimpleContent simpleContent = new SimpleContent();
+        complexType.setSimpleContent(simpleContent);
+        SimpleExtensionType simpleContentExtension = new SimpleExtensionType();
+        QName extensionBase = SchemaTypeConversion.convertType(type, false);
+        simpleContentExtension.setBase(extensionBase);
+        simpleContent.setExtension(simpleContentExtension);
+
+        Attribute valueAttribute = createAttribute(ATTRIBUTE_NAME_VALUE, type, true, true);
+        simpleContentExtension.getAttributeOrAttributeGroup().add(valueAttribute);
+
+        return complexType;
+    }
+
+    private DataType getGenericType(DataType type)
+    {
+        return ArrayUtils.isEmpty(type.getGenericTypes()) ? type : type.getGenericTypes()[0];
```

Comment:
BEcause it's used to process Lists so you only care about the first one... I'll change it to getFirstGenericType